### PR TITLE
feat(datepicker): add ivy support

### LINKED ISSUE #5257: ngx-bootstrap isn't working with ivy
### Bug description or feature request:
when building the angular application `ng build` with ivy enabled it throw the following error
> ERROR in Expected array when reading property declarations

for any package i'm using (tried alert, tabs and pagination)
i think (but not sure) the problem is with the datepicker
because before throwing the error it shows
`Compiling ngx-bootstrap/datepicker : module as esm5`

### StackBlitz that reproduces the issue:
using the starter templates without any edit other than making ivy enabled will throw the error

StackBlitz: https://stackblitz.com/edit/ngx-bootstrap?file=app%2Fapp.module.ts

### Versions of ngx-bootstrap, Angular, and Bootstrap:

ngx-bootstrap: 4.2.0

Angular: 8.0.0

Bootstrap: NA

### Build system: Angular CLI

### PR DESCRIPTION
**Pull request by Domainv**
Steps to reproduce:
ng new hello-ivy --enable-ivy
npm i
npm i ngx-bootstrap@next
npm run build

Closes #5257

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [ ] added/updated tests.
 - [ ] added/updated API documentation.
 - [ ] added/updated demos.

### CODE REVIEW

**Comment by ludmilanesvitiy** [comment]
`import { BsDatepickerModule } from 'ngx-bootstrap/datepicker';` - works well
but with  `import { BsDatepickerModule } from 'ngx-bootstrap';` - error thrown:
ERROR in Symbol BsDatepickerContainerComponent declared in /home/vs/Projects/testNGXIvy/node_modules/ngx-bootstrap/datepicker/themes/bs/bs-datepicker-container.component.d.ts is not exported from ngx-bootstrap (import into /home/vs/Projects/testNGXIvy/src/app/app.component.ts)
Looks like this is important issue, because a lot of users using just import from base module

**Comment by ludmilanesvitiy** [comment]
Tested with   "angularCompilerOptions": {    "enableIvy": false, and true, tested imports from ngx-bootstrap and ngx-bootstrap/module, smoke tested demo site - looks good.

File: src/datepicker/bs-datepicker.module.ts [hunk 1]
```diff
@@ -50,22 +37,35 @@ const _exports = [
...
     BsTimepickerViewComponent,
-
     BsCalendarLayoutComponent,
     BsDaysCalendarViewComponent,
     BsMonthCalendarViewComponent,
     BsYearsCalendarViewComponent,
-
     BsCustomDatesViewComponent,
-
-    ..._exports
```
Comment (Splaktar) [inline src/datepicker/bs-datepicker.module.ts:61]: Did you open (or find) an issue against ngcc with Ivy to support the spread operator?

Comment (Domainv) [reply src/datepicker/bs-datepicker.module.ts:61]: It already fixed, I checked it on angular 8.1.2
https://github.com/angular/angular/issues/29835

**Pull request merged**